TTS/tts/models/vits.py

```diff
@@ -396,13 +555,18 @@ def forward(
             - speaker_ids: :math:`[B]`
         """
         outputs = {}
-        sid, g = self._set_cond_input(aux_input)
-        x, m_p, logs_p, x_mask = self.text_encoder(x, x_lengths)
-
+        sid, g, lid = self._set_cond_input(aux_input)
         # speaker embedding
-        if self.num_speakers > 1 and sid is not None:
+        if self.args.use_speaker_embedding and sid is not None and not self.use_d_vector:
             g = self.emb_g(sid).unsqueeze(-1)  # [b, h, 1]
 
+        # language embedding
+        lang_emb = None
+        if self.args.use_language_embedding and lid is not None:
+            lang_emb = self.emb_l(lid).unsqueeze(-1)
```

Comment:
it is better to  check against `hasattr(self, "emb_l")` to not to miss broken config

Reply:
Should we change to that check for `if self.args.use_speaker_embedding and sid is not None:` and inference ?
Also maybe it's better not to check for `lid is not None`, that way the crash is more explicit (unexpected None instead of a shape error in the encoder) what do you guys think?